tools/lsp/common.rs

```diff
@@ -25,32 +28,142 @@ pub type UrlVersion = Option<i32>;
 #[cfg(target_arch = "wasm32")]
 use crate::wasm_prelude::*;
 
-pub fn uri_to_file(uri: &lsp_types::Url) -> Option<PathBuf> {
-    let path = uri.to_file_path().ok()?;
-    let cleaned_path = i_slint_compiler::pathutils::clean_path(&path);
-    Some(cleaned_path)
+pub fn uri_to_file(uri: &Url) -> Option<PathBuf> {
+    if uri.scheme() == "builtin" {
+        let path = String::from("builtin:") + uri.path();
```

Comment:
why not uri.to_string() ?

Reply:
Yeah, because you do wierd stuff in the other function. The URL should be `builtin:/foo/bar.slint`